### PR TITLE
Document NXDL file organisation

### LINKED ISSUE #914: Document how to associate name of NXDL with requirements in the XML file.
The manual must make clear how to properly construct an NXDL file that connects the NXDL file name with the `definition@name=".."` attribute in the top-most element and the required `enumeration` element in the `definition` field below.  This same name (and spelling, and upper/lower case representation) should match the file name.  Consider putting this into the manual as a sidebar.

And, any content referenced by the NXDL (such as images) be placed in a subdirectory with the name chosen by stripping off the `NX`.  (as in:  NXDL for `NXthing` would be in file `NXthing.nxdl.xml`)

file: `NXthing.nxdl.xml`
resources:  `thing/image1.png`

In the NXDL file:

```xml
<definition name="NXthing" ... />

<!-- later -->

        <field name="definition">
            <doc>Official NeXus NXDL schema to which this file conforms.</doc>
            <enumeration>
                <item value="NXthing"/>                
            </enumeration>
        </field>
```

_Originally posted by @prjemian in https://github.com/nexusformat/definitions/issues/910#issuecomment-809535008_

### PR DESCRIPTION
**Pull request by benajamin**
- fixes #914

### CODE REVIEW

**Comment by benajamin** [comment]
I think the current commit is a good basis for the task, but I'm sure there will be tweaks we can make to ensure that we are efficiently communicating the correct information with sufficient emphasis.

**Review by PeterC-DLS — APPROVED** [review]
LGTM

File: manual/source/nxdl-files.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+======================
+NXDL File Organisation
+======================
+
+NXDL File Name
+##############
+
+In order for the XML machinery to find and link the code in the various files, the name of the file must be composed of the definition name (matching both the spelling and the case) and a ".xml" extension. For example, the base class ``NXarbitrary_example`` should be defined by NXDL code within the ``NXarbitrary_example.xml`` file. Note also that the definition name is stated twice in application definitions, once in the ``definition`` tag, and again as the value of an ``item`` contained within the ``field`` tag that is named "definition".
```
Comment (prjemian) [inline manual/source/nxdl-files.rst:8]: change

```xml
and a ".xml" extension. For example, the base class ``NXarbitrary_example`` should be defined by NXDL code within the ``NXarbitrary_example.xml``
```

to

```xml
and a ".nxdl.xml" extension. For example, the base class ``NXarbitrary_example`` should be defined by NXDL code within the ``NXarbitrary_example.nxdl.xml``
```

File: manual/source/nxdl-files.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+         <enumeration>
+            <item value="NXarbitrary_example"/>                
+         </enumeration>
+      </field>
+   </definition>
+
+Documentation Images
+####################
+
+Including images in the documentation of NXDL definiitions can be very effective. The image files must go into a directory having the same name as the definition without the NX prefix.
```
Comment (prjemian) [inline manual/source/nxdl-files.rst:28]: change

```xml
Including images in
```

to

```xml
Including images (or other related files) in
```

**Review by RussBerg — APPROVED** [review]
LGTM

File: manual/source/nxdl-files.rst (hunk 2 above)
Comment (benajamin) [reply manual/source/nxdl-files.rst]: done

File: manual/source/nxdl-files.rst (hunk 1 above)
Comment (benajamin) [reply manual/source/nxdl-files.rst]: Good catch! Done.

**Pull request merged**